app/lib/search_query_transformer.rb

```diff
@@ -224,6 +224,14 @@ def language_code_from_term(term)
     end
   end
 
+  class TermValidator
+    def self.validate_date!(value)
+      return value if (value.match?(/\A\d{4}-\d{2}-\d{2}(\s\d{2}:\d{2})?\z/) || value.match?(/\A\d{1,11}\z/))
```

Comment:
the first regex here also includes a time which still throws an exception from ES, but the time was included to not break this test: https://github.com/mastodon/mastodon/blob/edeae945c0c9d6318d489ab720587621545063d0/spec/lib/search_query_transformer_spec.rb#L71-L79

ideally it seems like that test should be cleaned up, unless there's another non-ES search backend that does support that syntax?

Reply:
I think the point of the test is to confirm that the prefix andthe  phrase with spaces in it are parsed correctly as a single unit. It could be changed into something else.

Reply:
ah ok, I broke it out into two specs: one with quotes and a space and one for parsing dates. and updated the regex here to only match `yyyy-MM-dd`